package/MDAnalysis/lib/distances.py

```diff
@@ -455,8 +452,8 @@ def capped_distance(reference: Union[np.ndarray, 'AtomGroup'],
     # The check_coords decorator made sure that reference and configuration
     # are arrays of positions. Mypy does not know about that so we have to
     # tell it.
-    reference_positions: np.ndarray = reference  # type: ignore
-    configuration_positions: np.ndarray = configuration  # type: ignore
+    reference_positions: npt.NDArray = reference  # type: ignore
+    configuration_positions: npt.NDArray = configuration  # type: ignore
```

Comment:
Do we want to have subtyping on eg `npt.NDArray[np.float32]` ?

Reply:
Unsure, but I'd probably ask that this get punted to a separate issue (unless these changes lessen the quality of typing from what it used to be)?

I'd prefer not move the goalpost here from "bumping up our numpy requirements" to "incidentally dealing with a ton of typing issues"

Reply:
Yep totally fair. I'll raise an issue once merged. The typing here is improvement for sure, it can just be more stringent but agree its a separate issue.